shared/actions/platform-specific/index.native.tsx

```diff
@@ -458,6 +462,86 @@ function* requestContactPermissions(
   ])
 }
 
+async function manageContactsCache(
+  state: TypedState,
+  action: SettingsGen.LoadedContactImportEnabledPayload | ConfigGen.MobileAppStatePayload,
+  logger: Saga.SagaLogger
+) {
+  if (action.type === ConfigGen.mobileAppState && action.payload.nextAppState !== 'active') {
+    return
+  }
+
+  if (state.settings.contacts.importEnabled === false) {
+    return RPCTypes.contactsSaveContactListRpcPromise({contacts: []}).then(() =>
+      SettingsGen.createSetContactImportedCount({count: null})
+    )
+  }
+
+  // get permissions if we haven't loaded them for some reason
+  let {permissionStatus} = state.settings.contacts
+  if (permissionStatus === 'unknown') {
+    permissionStatus = await loadContactPermissionFromNative()
+  }
+  const perm = permissionStatus === 'granted'
+
+  const enabled = state.settings.contacts.importEnabled
+  if (!enabled || !perm) {
+    if (enabled && !perm) {
+      logger.info('contact import enabled but no contact permissions')
+    }
+    if (enabled === null) {
+      logger.info("haven't loaded contact import enabled")
+    }
+    return
+  }
+
+  // feature enabled and permission granted
+  const contacts = await Contacts.getContactsAsync()
+  const mapped = contacts.data.reduce((ret: Array<RPCTypes.Contact>, contact) => {
+    const {name, phoneNumbers = [], emails = []} = contact
```

Comment:
can these be null vs undefined

Reply:
Docs say they can't be null or undefined, always an array